src/platform/loader-node.ts

```diff
@@ -19,12 +19,21 @@ export class Loader extends LoaderBase {
     return new Loader(this.urlMap);
   }
   async loadFile(path: string): Promise<string> {
-    return this.loadFileData(path, 'utf-8') as Promise<string>;
+    const data = this.loadFileData(path, 'utf-8');
+    if (typeof data !== 'string') {
+      throw new Error(`loadFileData returned non-String for utf-8 file [${path}]`);
+    }
+    return data;
   }
   async loadBinaryFile(path: string): Promise<ArrayBuffer> {
-    return this.loadFileData(path) as Promise<ArrayBuffer>;
+    const data = await this.loadFileData(path);
+    if (!(data instanceof Buffer)) {
+      throw new Error(`loadFileData returned non-Buffer for binary file [${path}]`);
+    }
+    // convert Buffer to ArrayBuffer (slice in case a small Buffer is a view on a shared ArrayBuffer)
+    return data.buffer.slice(data.byteOffset, data.byteOffset + data.byteLength);
   }
-  private async loadFileData(path: string, encoding?: string): Promise<string | ArrayBuffer> {
+  private async loadFileData(path: string, encoding?: string): Promise<string | Buffer> {
     return new Promise((resolve, reject) => {
       fs.readFile(path, encoding, (err, data: string) => {
```

Comment:
I think this might need to be `options: {encoding}`, but I'm not sure what the correct usage is.

Reply:
Seems that did not help. We're still getting an unexpected type returned.

Reply:
From the node docs: `If options is a string, then it specifies the encoding:`

Reply:
Hmm, I'm working from the node_modules/@types/node/fs.d.ts declaration.
```
function readFile(path: PathLike | number, options: { encoding: string; flag?: string; } | string, callback: (err: NodeJS.ErrnoException | null, data: string) => void): void;
```
I'm not sure what to make of the mismatch, maybe I should invoke the Cromwellian hammer (nuke node_modules) and see if mine is just old?

The version in `@types/node/package.json` is
```
  "version": "10.17.4"
```
But there are some other version numbers in there which I think suggest that the types package is for Node v10.
```
  "_from": "@types/node@^10.1.0",
  "_id": "@types/node@10.17.4",
  "typeScriptVersion": "2.0",
```

https://nodejs.org/api/fs.html#fs_fs_readfile_path_options_callback
Interestingly, the doc suggests that both usages are valid in Node v13.

Reply:
Seems like `options: { encoding: string; flag?: string; } | string` is not incompatible. Which is to say, it's either an object, or a string (and if it's a string, this is used as `encoding`).